File: cakk-batch/src/main/java/com/cakk/batch/config/SlackWebhookConfig.java
package com.cakk.batch.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import net.gpedro.integrations.slack.SlackApi;

@Configuration
public class SlackWebhookConfig {

	private final String slackWebhookUrl;

	public SlackWebhookConfig(@Value("${slack.webhook.url}") String slackWebhookUrl) {
		this.slackWebhookUrl = slackWebhookUrl;
	}

	@Bean
	public SlackApi slackApi() {
		return new SlackApi(slackWebhookUrl);
	}
}

File: cakk-external/src/main/kotlin/com/cakk/external/config/SlackWebhookConfig.kt
package com.cakk.external.config

import net.gpedro.integrations.slack.SlackApi

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class SlackWebhookConfig(
	@Value("\${slack.webhook.url}")
	private val slackWebhookUrl: String
) {

	@Bean
	fun slackApi(): SlackApi {
		return SlackApi(slackWebhookUrl)
	}
}
